sdk/core/azure-json/src/main/java/com/azure/json/package-info.java

```diff
@@ -2,6 +2,499 @@
+ * {@link com.azure.json.JsonSerializable#fromJson(com.azure.json.JsonReader) fromJson(JsonReader)} that defines how to
+ * read an instance of the object from JSON, being read from the {@link com.azure.json.JsonReader}. The default
+ * implementation of {@link com.azure.json.JsonSerializable#fromJson(com.azure.json.JsonReader) fromJson(JsonReader)}
+ * throws an {@link java.lang.UnsupportedOperationException} if the static method isn't overwritten. Given that the
+ * type itself manages JSON serialization the type can be fluent, immutable, or a mix of fluent and immutable, it 
+ * doesn't matter as all logic is self-encapsulated.</p>
+ *
+ * <p><strong>Sample: Fluent JsonSerializable</strong></p>
+ *
+ * <!-- src_embed com.azure.json.JsonSerializable.fluent -->
+ * <pre>
+ *
+ * &#47;**
+ *  * Implementation of JsonSerializable where all properties are fluently set.
+ *  *&#47;
+ * public class FluentJsonSerializableExample implements JsonSerializable&lt;FluentJsonSerializableExample&gt; &#123;
+ *     private int anInt;
+ *     private boolean aBoolean;
+ *     private String aString;
+ *     private Double aNullableDecimal;
+ *
+ *     &#47;**
+ *      * Sets an integer value.
+ *      *
+ *      * &#64;param anInt The integer value.
+ *      * &#64;return The update FluentJsonSerializableExample
+ *      *&#47;
+ *     public FluentJsonSerializableExample setAnInt&#40;int anInt&#41; &#123;
+ *         this.anInt = anInt;
+ *         return this;
+ *     &#125;
+ *
+ *     &#47;**
+ *      * Sets a boolean value.
+ *      *
+ *      * &#64;param aBoolean The boolean value.
+ *      * &#64;return The update FluentJsonSerializableExample
+ *      *&#47;
+ *     public FluentJsonSerializableExample setABoolean&#40;boolean aBoolean&#41; &#123;
+ *         this.aBoolean = aBoolean;
+ *         return  this;
+ *     &#125;
+ *
+ *     &#47;**
+ *      * Sets a string value.
+ *      *
+ *      * &#64;param aString The string value.
+ *      * &#64;return The update FluentJsonSerializableExample
+ *      *&#47;
+ *     public FluentJsonSerializableExample setAString&#40;String aString&#41; &#123;
+ *         this.aString = aString;
+ *         return this;
+ *     &#125;
+ *
+ *     &#47;**
+ *      * Sets a nullable decimal value.
+ *      *
+ *      * &#64;param aNullableDecimal The nullable decimal value.
+ *      * &#64;return The update FluentJsonSerializableExample
+ *      *&#47;
+ *     public FluentJsonSerializableExample setANullableDecimal&#40;Double aNullableDecimal&#41; &#123;
+ *         this.aNullableDecimal = aNullableDecimal;
+ *         return this;
+ *     &#125;
+ *
+ *     &#64;Override
+ *     public JsonWriter toJson&#40;JsonWriter jsonWriter&#41; throws IOException &#123;
+ *         &#47;&#47; Optionally 'flush' can be included in the method call chain. It isn't necessary as the caller into this
+ *         &#47;&#47; method should handle flushing the JsonWriter.
```

Comment:
Don't give me 'optionally' - it doesn't help me as a reader as to whether I should call this or not.